starters/express-typescript-typeorm-postgres/src/utils/log-helper.ts

```diff
@@ -0,0 +1,72 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+const allowedLogLevels = process.env.ALLOWED_LOG_LEVELS || '[]';
+
+type LogLevel = 'INFO' | 'DEBUG' | 'WARN' | 'ERROR';
+
+let LOG_LEVELS: LogLevel[] = ['DEBUG', 'INFO', 'WARN', 'ERROR'];
+try {
+  const parsedLogLevels = JSON.parse(allowedLogLevels);
+  if (parsedLogLevels.length !== 0) {
+    LOG_LEVELS = parsedLogLevels;
+  }
+} catch (e) {
+  console.warn(
+    `[WARN] Could not parse ALLOWED_LOG_LEVELS env variable, logger defaults to all levels`,
+    e
+  );
+}
+
+class Logger {
+  private enabledLevels: Set<LogLevel>;
+
+  constructor(allowedLogLevels: LogLevel[]) {
+    this.enabledLevels = new Set(allowedLogLevels);
```

Comment:
One concern to note, which decided how succinct the code can be, is whether or not we want to support the case where `process.env.ALLOWED_LOG_LEVELS` changing after startup. Assuming we do we, we should probably check the `process.env.ALLOWED_LOG_LEVELS` every time.

Reply:
I thought about it, and my recent experience with logging and bringing down a whole production AWS infrastructure by adding more and more logic to logging says that we should not allow that, but if somebody needs that functionality in their app, they can certainly implement it building on top of our code.

**My reasoning:**
Even if we simplify the environment variables to be a comma separated list (I'll get to it soon), it is two or more operations:
1. Read the env variable
2. Run `string.prototype.split()` on it
3. Iterate over the array to find if the wanted log-level is in there

Doing these 3 steps on every log has a toll on performance, even if only "Error" is allowed, because the above 3 steps will run on every log call no matter what. So a production code of:

```
function somethingHandler(req, res, next) {
  Logger.debug(req.url)
  let something = req.params.something
  if (!something) {
    Logger.warn(`${req.params.something} is undefined, method was called from an app below version 3.2. Defaulting to "whatever" value`
    something = "whatever
  }
  // ...
  
}
```

Would do those operations even if we don't log anything because of the log levels. That is why I'd avoid putting too much logic into each logger call.